awriter/writer.go

```diff
@@ -211,6 +228,13 @@ func writeHeader(tw *tar.Writer, devices []string, name string,
 		return errors.New("writer: can not store header-info")
 	}
 
+	// write scripts
+	if scr != nil {
+		if err := writeScripts(tw, scr); err != nil {
+			return err
+		}
+	}
+
```

Comment:
I don't think this needs to be protected here. The better place would be in `artifacts.go` where you are writing artifact reading command line parameters. I will add a check there.